features/ftue/impl/src/main/kotlin/io/element/android/features/ftue/impl/sessionverification/FtueSessionVerificationFlowNode.kt

```diff
@@ -83,7 +88,12 @@ class FtueSessionVerificationFlowNode @AssistedInject constructor(
                     .params(SecureBackupEntryPoint.Params(SecureBackupEntryPoint.InitialTarget.EnterRecoveryKey))
                     .callback(object : SecureBackupEntryPoint.Callback {
                         override fun onDone() {
-                            callback.onDone()
+                            lifecycleScope.launch {
+                                // Mark the session as verified
+                                sessionVerificationService.skipVerification()
```

Comment:
It's weird to me to see a call to `sessionVerificationService.skipVerification()` here. The service should know when the session is verified, there should be no need to call `skipVerification` here (and it's not skipping, since it's done IIUC).

Reply:
That only works for session verification through emoji, since we have the `didFinish` callback, but it won't work for verification through recovery key.

We need something else to mark the session as verified. Since the logic is exactly the same as `skipVerification` I could just rename it to something more generic, or I could add the same code in `SecureBackupEnterRecoveryKeyPresenter`. WDYT?

Reply:
I've done it in the presenter instead to avoid cyclic dependencies between `SessionVerificationService` and `EncryptionService`.

Reply:
I do not understand, isn't it possible to inject `private val sessionStore: SessionStore` to `RustEncryptionService`?

And write:

```kotlin
    override suspend fun recover(recoveryKey: String): Result<Unit> = withContext(dispatchers.io) {
        runCatching {
            service.recover(recoveryKey)
            val existingSession = sessionStore.getSession(client.userId())
                ?: error("Failed to save verification state. No session with id ${client.userId()}")
            sessionStore.updateData(existingSession.copy(needsVerification = false))
        }.mapFailure {
            it.mapRecoveryException()
        }
    }
```

Reply:
I wanted to reuse the method in `SessionVerificationService` to centralise this kind of change, but I can do it like this too.